### PR TITLE
arianna: init at 1.0.0

### PR DESCRIPTION
**Pull request by apfelkuchen6**
###### Description of changes

Adds an ebook reader for KDE. 
Project url: https://invent.kde.org/graphics/arianna

Requires kirigami-addons 0.8.0 (#226156) for the settings-panel to work.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/office/arianna/default.nix
Comment (Rampoina) [inline pkgs/applications/office/arianna/default.nix:34]: Why not `fetchgit` from https://invent.kde.org/graphics/arianna

```suggestion
  src = fetchgit {
    url = "https://invent.kde.org/graphics/arianna";
    rev = "v${version}";
    sha256 = "sha256-X3PDGWsQ8Alj5fisZC1tTHQDLPmjtiLw0X9gMvh5KFI=";
  };
```

**Review by Rampoina — CHANGES_REQUESTED** [review]
Thanks for the package!
- [x] Tested basic functionality of all binary files (usually in ./result/bin/)
- Closes #226403

File: pkgs/applications/office/arianna/default.nix
Comment (apfelkuchen6) [reply pkgs/applications/office/arianna/default.nix]: oops, fixed

File: pkgs/applications/office/arianna/default.nix
Comment (apfelkuchen6) [reply pkgs/applications/office/arianna/default.nix:33]: `fetchgit` is generally slower and has more overhead than fetching tarballs. This is one of the main reasons why `fetchFromGitHub` and `fetchFromGitLab` exist (they download tarballs and don't use git).

Is there any reason not to use the well-mirrored official release tarballs?

Comment (Rampoina) [reply pkgs/applications/office/arianna/default.nix:33]: I'm not sure which one is the preferred way really, which is why I was asking :)
I'll leave it to somebody more experienced in NixOS to comment on.

**Review by felschr — CHANGES_REQUESTED** [review]
Fails to start when running in Gnome:
```console
❯ arianna
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
QApplication: invalid style override 'adwaita' passed, ignoring it.
        Available styles: Windows, Fusion
QQmlApplicationEngine failed to load component
qrc:/content/ui/main.qml:31:22: Type LibraryPage unavailable
qrc:/content/ui/LibraryPage.qml:10:1: module "org.kde.quickcharts" is not installed
```

**Comment by felschr** [comment]
It starts now, but it doesn't show any cover images for any of my epubs.

image

It does log some errors for saving cover images:

<details><summary>Console output</summary>
<p>

```console
❯ arianna 
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
QApplication: invalid style override 'adwaita' passed, ignoring it.
        Available styles: Windows, Fusion
search started
Unsupported metadata tag "description"
org.kde.arianna: Error saving image "/home/felschr/.cache/arianna/covers/9781101658055.jpg"
Unsupported metadata tag "description"
org.kde.arianna: Error saving image "/home/felschr/.cache/arianna/covers/9781101157879.jpg"
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/9781101658055.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/9781101157879.jpg
org.kde.arianna: Error saving image "/home/felschr/.cache/arianna/covers/urn_uuid_5ca1d576-8986-4c35-abda-f1cc7962bb90.jpg"
org.kde.arianna: Error saving image "/home/felschr/.cache/arianna/covers/urn_uuid_d5070ed1-fad9-48cc-a871-36c1e4e3c976.jpg"
Broken filename passed to function
Broken filename passed to function
org.kde.arianna: Error saving image "/home/felschr/.cache/arianna/covers/9781733672719\u00009781733672719.jpg"
Broken filename passed to function
Broken filename passed to function
Broken filename passed to function
org.kde.arianna: Error saving image "/home/felschr/.cache/arianna/covers/urn_uuid_aa45d95f-118a-408a-a842-bffad5927acd.jpg"
Unsupported metadata tag ""
Asked for unknown item "" ("id-idm140737313764736", "id-idm140737313714080", "html-css", "id-idm140737314091408", "id-idm140737314618352", "
id-idm140737313635344", "id-idm140737313582336", "id-idm140737314362032", "id-idm140737313518704", "ncx", "id-idm140737313933920", "id-idm14
0737313825664", "id-idm140737314262784", "id-idm140737314843680", "id-idm140737313988192", "id-idm140737314797136", "id-idm140737314808400",
 "id-idm140737313396464", "id-idm140737313468176", "htmltoc", "id-idm140737314686288", "id-idm140737314529920", "docbook-css", "id-idm140737
314158304", "id-idm140737314432192", "id-idm140737314039472")
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/9781733672719%00978173367
2719.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/urn_uuid_5ca1d576-8986-4c
35-abda-f1cc7962bb90.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/urn_uuid_d5070ed1-fad9-48
cc-a871-36c1e4e3c976.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/urn_uuid_aa45d95f-118a-40
8a-a842-bffad5927acd.jpg
Broken filename passed to function
Broken filename passed to function
Broken filename passed to function
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/9781101658055.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/9781101157879.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/urn_uuid_5ca1d576-8986-4c
35-abda-f1cc7962bb90.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/urn_uuid_d5070ed1-fad9-48
cc-a871-36c1e4e3c976.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/9781733672719%00978173367
2719.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/urn_uuid_aa45d95f-118a-40
8a-a842-bffad5927acd.jpg
Broken filename passed to function
Broken filename passed to function
Broken filename passed to function
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/9781733672719%00978173367
2719.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/9781101658055.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/9781101157879.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/urn_uuid_aa45d95f-118a-40
8a-a842-bffad5927acd.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/urn_uuid_5ca1d576-8986-4c
35-abda-f1cc7962bb90.jpg
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/felschr/.cache/arianna/covers/urn_uuid_d5070ed1-fad9-48
cc-a871-36c1e4e3c976.jpg
```

</p>
</details>

**Comment by felschr** [comment]
Also, looking at the official screenshots, the icons are missing.

**Comment by apfelkuchen6** [comment]
After adding a `wrapGAppsHook`, the thumbnails also work on Gnome:
gnome

**Comment by felschr** [comment]
Hmm, the icons & thumbnails are still missing for me.
It seems that everything else works, though. I can open & read ebooks without problems.

**Comment by apfelkuchen6** [comment]
Closing in favor of #227816.

**Comment by Thra11** [comment]
Apologies, I forgot to check for existing pull requests before creating my own. Mine doesn't have `wrapGAppsHook`. Do you think it should have it? Also, mine fetches the sources from invent.kde.org gitlab instead of using the KDE mirror.

I also had an issue with it writing and reading the cover cache at first. Works now, though I'm not sure if it was solved by adding a missing dependency or whether there might be an upstream bug where it doesn't create the cache directory until later than it should do.

If you have time, it would be helpful if you can test my PR and check whether it suffers from any of the issues discussed in this one.

Feel free to add yourself to maintainers on my version if you like.

**Comment by apfelkuchen6** [comment]
I'm not completely sure whether a `wrapGAppsHook`  (and/or possibly something else) is needed here. I just know that Qt internally uses Gtk on some systems for some things, so adding the hook was kind of a shot in the dark that turned out to work (on first launch) in my fresh gnome-vm (`nixos-rebuild build-vm -I nixos-config=gnome-vm.nix -I nixpkgs=.` with
```
{ pkgs, ... }:{
  environment.systemPackages = [ pkgs.arianna ];
  services.xserver.desktopManager.gnome.enable = true;
  services.xserver.enable = true;
  users.users.foo = {
    isNormalUser = true;
    password = "foo";
  };
}
```
). But since this doesn't work for @felschr, this might need further investigation. I guess it's possible that thumbnail generation on Gnome only works with the hook, but if it failed once it isn't attempted again, but it also might be something else entirely.

I will try your PR later.

**Comment by felschr** [comment]
With or without `wrapGAppsHook` didn't make a difference for me at least. @apfelkuchen6, did it make a difference in your VM?
I'm getting the same issues of missing cover images & icons I've mentioned in the other PR with this one.

Could even be some issue unrelated to Gnome and/or an issue with upstream itself.
Not sure what anyone else thinks, but if this works fine on KDE I think it's still good to be merged.

**Comment by felschr** [comment]
With or without `wrapGAppsHook` didn't make a difference for me at least. @apfelkuchen6, did it make a difference in your VM?
I'm getting the same issues of missing cover images & icons I've mentioned here with the new PR.

Could even be some issue unrelated to Gnome and/or an issue with upstream itself.
Not sure what anyone else thinks, but if this works fine on KDE I think it's still good to be merged.

**Comment by Thra11** [comment]
Haven't tested this rigorously yet, but anecdotally…
1. Run arianna for the first time on a clean system/vm and add a book (either put the book somewhere baloo can find it, or manually add it in arianna -> book cover doesn't show, message in stdout:
```
qrc:/content/ui/GridBrowserDelegate.qml:117:13: QML Image: Cannot open: file:///home/thrall/.cache/arianna/covers/9781580805193%00b3235430-024c-4f9c-babb-9d95a0ba4e7c.jpg
```
cache directory (`~/.cache/arianna/covers`) doesn't exist.

2. Close arianna
3. Reopen arianna and add a second book (again, either place a new file where baloo can find it or add manually) -> second book's cover appears in arianna. No warning in stdout. All books added afterwards show their covers.

The first book never loads its cover unless you remove and re-add it.

**Comment by Thra11** [comment]
(I reported it as a bug upstream https://bugs.kde.org/show_bug.cgi?id=469138)

**Pull request closed without merging**
